### PR TITLE
Control panel styles and parts

### PR DESCRIPTION
**Pull request by mirisuzanne**
## Related Issue(s)
_Reminder to add related issue(s), if available._

- #9 
- #5 

I provided some basic styles for the control panel, but they could be improved. I'm also not sure if we want to make the entire panel replaceable, or if it's enough to replace the controls inside the panel. I suppose the advantage of replaceing the entire panel is that it could be more easily styled by the page.

### CODE REVIEW

**Comment by mirisuzanne** [comment]
@stacyk this isn't ready for a final review, but I could use your feedback on it before going further.

**Comment by mirisuzanne** [comment]
Discussion this morning:
- Make the entire control panel replaceable
- Expose button parts for light dom style
- Provide some default styles in the light dom css theme

**Comment by mirisuzanne** [comment]
@jamesnw @jerivas @stacyk This should be ready for review, and makes some of the changes we discussed this morning. The separate `slide-canvas`/`slide-note` bits are part of #11 - which we can clean up once this is merged.

**Review by stacyk — COMMENTED** [review]
I'm just starting to look at this PR, but we have some issues in Safari. I'll keep poking around but wanted to post this in case you had any ideas what is happening here @mirisuzanne. 


Screenshot 2024-01-03 at 10 36 48 PM

**Comment by stacyk** [comment]
@mirisuzanne Maybe we can use Auto-fill instead?

autofill

**Comment by jamesnw** [comment]
I like this! I needed to make some minor adjustments to get the control panel working- you can see those changes in https://github.com/oddbird/slide-deck/pull/10/commits/5958c3a6cc7a90c9f22ebb5faa92261d1189729f

Notably-
- I needed to set `this.controlPanel` to the slotted element, falling back to the shadow dom
- I changed the slideCount from `this.childElementCount` to `this.querySelectorAll(':scope > :not([slot])').length`

I also added an [example](https://deploy-preview-10--slide-deck.netlify.app/custom-panel#slide_my-slides-1), which is fairly barebones at this point. We probably also want to document this in the README?

File: slide-deck.js [hunk 1]
```diff
@@ -210,17 +132,18 @@ class slideDeck extends HTMLElement {
...
 
     // relevant nodes
     this.body = document.querySelector('body');
-    this.controlPanel = this.shadowRoot.querySelector(`[part="controls"]`);
+    this.controlPanel = this.querySelector(`[slot="control-panel"]`) ??
+      this.shadowRoot.querySelector(`[part="control-panel"]`);
 
     // initial setup
-    this.slideCount = this.childElementCount;
+    const slides = this.querySelectorAll(':scope > :not([slot])');
```
Comment (mirisuzanne) [inline slide-deck.js:143]: @jamesnw what do you think about making this a bit more specific?

```suggestion
    const slides = this.querySelectorAll(':scope > :not([slot=control-panel])');
```

**Comment by mirisuzanne** [comment]
@stacyk I think auto-fill works ok for this. That seems to be a known Safari bug with auto-fit and containers (If you remove the `container` property on `slide-deck`, it works fine):

https://bugs.webkit.org/show_bug.cgi?id=256047

File: slide-deck.js (hunk 1 above)
Comment (jamesnw) [reply slide-deck.js:143]: @mirisuzanne I intentionally made that more generic. I don't think we have plans for additional slots, but this future proofs it slightly. I think it also matches the browser behavior a bit closer- Slides are the direct children in the default slot, and the default slot contains everything that is not slotted.

Alternatively, could we query the shadow dom default slot content directly? Then we are directly measuring what we want, but I'm not quickly seeing how to do that.

Comment (mirisuzanne) [reply slide-deck.js:143]: That makes sense, let's leave it as-is for now.

**Review by jamesnw — APPROVED** [review]
🚀

**Pull request merged**